extractor/src/test/java/org/schabi/newpipe/extractor/services/bandcamp/BandcampSearchExtractorTest.java

```diff
@@ -0,0 +1,105 @@
+     * the accordingly named song by Zach Benson
+     */
+    @Test
+    public void testStreamSearch() throws ExtractionException, IOException {
+        SearchExtractor extractor = bandcamp.getSearchExtractor("best friend's basement");
+
+        ListExtractor.InfoItemsPage<InfoItem> page = extractor.getInitialPage();
+        InfoItem bestFriendsBasement = page.getItems().get(0);
+
+        // The track by Zach Benson should be the first result, no?
+        assertEquals("Best Friend's Basement", bestFriendsBasement.getName());
+        assertTrue(bestFriendsBasement.getThumbnailUrl().endsWith(".jpg"));
+        assertTrue(bestFriendsBasement.getThumbnailUrl().contains("f4.bcbits.com/img/"));
+        assertEquals(InfoItem.InfoType.STREAM, bestFriendsBasement.getInfoType());
+    }
+
+    /**
+     * Tests whether searching bandcamp for "C418" returns the artist's profile
+     */
+    @Test
+    public void testChannelSearch() throws ExtractionException, IOException {
+        SearchExtractor extractor = bandcamp.getSearchExtractor("C418");
+        InfoItem c418 = extractor.getInitialPage()
+                .getItems().get(0);
+
+        // C418's artist profile should be the first result, no?
+        assertEquals("C418", c418.getName());
+        assertTrue(c418.getThumbnailUrl().endsWith(".jpg"));
+        assertTrue(c418.getThumbnailUrl().contains("f4.bcbits.com/img/"));
+        assertEquals("https://c418.bandcamp.com", c418.getUrl());
+
+    }
+
+    /**
+     * Tests whether searching bandcamp for "minecraft volume alpha" returns the corresponding album
+     */
+    @Test
+    public void testAlbumSearch() throws ExtractionException, IOException {
+        SearchExtractor extractor = bandcamp.getSearchExtractor("minecraft volume alpha");
+        InfoItem minecraft = extractor.getInitialPage()
+                .getItems().get(0);
+
+        // Minecraft volume alpha should be the first result, no?
+        assertEquals("Minecraft - Volume Alpha", minecraft.getName());
+        assertTrue(minecraft.getThumbnailUrl().endsWith(".jpg"));
+        assertTrue(minecraft.getThumbnailUrl().contains("f4.bcbits.com/img/"));
+        assertEquals("https://c418.bandcamp.com/album/minecraft-volume-alpha", minecraft.getUrl());
+
+        // Verify that playlist tracks counts get extracted correctly
+        assertEquals(24, ((PlaylistInfoItem) minecraft).getStreamCount());
+
+    }
+
+    /**
+     * Tests searches with multiple pages
+     */
+    @Test
+    public void testMultiplePages() throws ExtractionException, IOException {
+        // A query practically guaranteed to have the maximum amount of pages
+        SearchExtractor extractor = bandcamp.getSearchExtractor("e");
+
+        assertEquals("https://bandcamp.com/search?q=e&page=2", extractor.getInitialPage().getNextPageUrl());
+
+        assertEquals("https://bandcamp.com/search?q=e&page=3", extractor.getPage(extractor.getNextPageUrl()).getNextPageUrl());
+    }
+}
```

Comment:
Please implement this test in a similar way as `YoutubeSearchExtractorTest` currently does

Reply:
Should I remove my existing tests?

Reply:
See 8f6c00f8d81d16de1b2f08a72452ebc5ad03c88c